src/jsxc.lib.xmpp.js

```diff
@@ -1012,6 +1014,73 @@ jsxc.xmpp = {
       return true;
    },
 
+   onComposing: function(stanza) {
+      var composing = $(stanza).find("composing[xmlns='http://jabber.org/protocol/chatstates']");
+      if (composing.length === 0) {
+         return true;
+      }
+
+      var type = $(stanza).attr("type");
+      var from = $(stanza).attr("from");
+
+      // ignore own notifications in groupchat
+      if (type === 'groupchat' &&
+           Strophe.getResourceFromJid(from) === Strophe.getNodeFromJid(jsxc.xmpp.conn.jid)) {
+          return true;
+      }
+
+      var bid = jsxc.jidToBid(from);
+      var user = type === 'chat' ? Strophe.getNodeFromJid(from) : Strophe.getResourceFromJid(from);
+      $('#jsxc_windowList .jsxc_windowItem').each(function() {
+         var self = $(this);
+         var winBid = self.data('bid');
+
+         // check conversation
+         if (winBid === bid) {
```

Comment:
Is it not easier and faster to get only the one desired window? You could use `jsxc.gui.window.get(bid)`.